test/schema_test.exs

```diff
@@ -138,4 +146,60 @@ defmodule OpenApiSpex.SchemaTest do
     assert :ok = Schema.validate(schema, "bar", %{})
   end
 
+  test "Validate schema type object when value is array" do
+    schema = %Schema{
+      type: :object
+    }
+    assert {:error, _} = Schema.validate(schema, [], %{})
+  end
+
+  test "Validate schema type array when value is object" do
+    schema = %Schema{
+      type: :array
+    }
+    assert {:error, _} = Schema.validate(schema, %{}, %{})
+  end
+
+  test "Validate schema type boolean when value is object" do
+    schema = %Schema{
+      type: :boolean
+    }
+    assert {:error, _} = Schema.validate(schema, %{}, %{})
+  end
+
+  test "Validate schema type string when value is object" do
+    schema = %Schema{
+      type: :string
+    }
+    assert {:error, _} = Schema.validate(schema, %{}, %{})
+  end
+
+  test "Validate schema type string when value is DateTime" do
+    schema = %Schema{
+      type: :string
+    }
```

Comment:
For this to be valid, I think it would require a `format: :"date-time"` in the schema.